server/service/src/invoice_line/stock_in_line/update/validate.rs

```diff
@@ -51,6 +52,17 @@ pub fn validate(
     if !check_location_exists(connection, store_id, &input.location)? {
         return Err(LocationDoesNotExist);
     }
+    // TODO
+    match &input.item_variant_id {
+        Some(NullableUpdate {
+            value: Some(item_variant_id),
+        }) => {
+            if !check_item_variant_exists(connection, item_variant_id)? {
+                return Err(ItemVariantDoesNotExist);
+            }
+        }
+        _ => {}
+    }
```

Comment:
```suggestion
    match &input.item_variant_id {
        Some(NullableUpdate {
            value: Some(item_variant_id),
        }) => {
            if !check_item_variant_exists(connection, item_variant_id)? {
                return Err(ItemVariantDoesNotExist);
            }
        }
        _ => {} //  We don't need to check item_variant if it's not being update, or if it's being updated to None
    }
```

I feel like this is sufficiently complicated to justify a comment? Took me a bit to parse it, possibly dad brain slowness though haha.